### PR TITLE
Drop support for deleting conflicts through cli and add cardinality column for keyless table conflicts 

### PR DESCRIPTION
**Pull request by druvv**
Support for deleting conflicts "manually" through the cli has been dropped in favor of the equivalent SQL commands.

Previously you could delete a conflict by specifying the primary key of the conflict:
`dolt conflicts resolve some_table pk_col 1`

Now, you can only delete a conflict through the SQL path:
`dolt sql -q "DELETE from dolt_conflicts_some_table where pk_col = 1"`

In addition, this PR also adds a `cardinality` column to `dolt_conflicts_$tableName` for keyless source tables. `cardinality` is the number of rows with the specified column values.

For example:
```bash
$ dolt sql -q "SELECT base_c0, base_c1, base_cardinality, our_cardinality, their_cardinality from dolt_conflicts_dupe;"
+---------+---------+------------------+-----------------+-------------------+
| base_c0 | base_c1 | base_cardinality | our_cardinality | their_cardinality |
+---------+---------+------------------+-----------------+-------------------+
| 1       | 1       | 10               | 6               | 8                 |
+---------+---------+------------------+-----------------+-------------------+
```

This PR also contains changes that reimplements `dolt conflicts resolve --ours | --theirs` through the SQL path.

### CODE REVIEW

File: go/libraries/doltcore/sqle/sqlddl_test.go [hunk 1]
```diff
@@ -774,17 +773,17 @@ func TestRenameTableStatements(t *testing.T) {
...
 
 	var dEnv *env.DoltEnv
 	setup := func() {
 		dEnv = dtestutils.CreateTestEnv()
 		CreateTestDatabase(dEnv, t)
 
-		dtestutils.CreateTestTable(t, dEnv, "dolt_docs",
-			doltdocs.DocsSchema,
-			NewRow(types.String("LICENSE.md"), types.String("A license")))
+		//dtestutils.CreateTestTable(t, dEnv, "dolt_docs",
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/sqlddl_test.go:784]: Revert?

File: go/libraries/doltcore/sqle/sqldelete_test.go [hunk 2]
```diff
@@ -190,14 +189,14 @@ func TestExecuteDeleteSystemTables(t *testing.T) {
...
 var systemTableDeleteTests = []DeleteTest{
-	{
-		Name: "delete dolt_docs",
-		AdditionalSetup: CreateTableFn("dolt_docs",
-			doltdocs.DocsSchema,
-			NewRow(types.String("LICENSE.md"), types.String("A license"))),
-		DeleteQuery: "delete from dolt_docs",
-		ExpectedErr: "cannot delete from table",
-	},
+	//{
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/sqldelete_test.go:192]: ?

File: go/libraries/doltcore/sqle/sqlinsert_test.go [hunk 3]
```diff
@@ -377,14 +376,14 @@ func TestExecuteInsert(t *testing.T) {
...
 var systemTableInsertTests = []InsertTest{
-	{
-		Name: "insert into dolt_docs",
-		AdditionalSetup: CreateTableFn("dolt_docs",
-			doltdocs.DocsSchema,
-			NewRow(types.String("LICENSE.md"), types.String("A license"))),
-		InsertQuery: "insert into dolt_docs (doc_name, doc_text) values ('README.md', 'Some text')",
-		ExpectedErr: "cannot insert into table",
-	},
+	//{
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/sqlinsert_test.go:379]: ?

File: integration-tests/bats/keyless.bats [hunk 4]
```diff
@@ -827,7 +821,6 @@ SQL
 }
 
 @test "keyless: merge delete+add on two branches" {
-    skip_nbf_dolt_1 "conflicts resolve not implemented"
     dolt branch left
```
Comment (zachmu) [inline integration-tests/bats/keyless.bats:824]: Need a test in this file exercising the conflict cardinality logic you added

File: go/cmd/dolt/commands/cnfcmds/auto_resolve.go [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+type AutoResolveStrategy int
+
+const (
+	AutoResolveStrategyOurs AutoResolveStrategy = iota
+	AutoResolveStrategyTheirs
+)
+
+var ErrConfSchIncompatible = errors.New("the conflict schema's columns are not equal to the current schema's columns, please resolve manually")
+
+func AutoResolveAll(ctx context.Context, dEnv *env.DoltEnv, strategy AutoResolveStrategy) error {
```
Comment (zachmu) [inline go/cmd/dolt/commands/cnfcmds/auto_resolve.go:46]: Method docs for exported methods in this file

**Review by zachmu — APPROVED** [review]
LGTM, just a couple comments

**Comment by druvv** [comment]
@zachmu The dolt doc tests are commented out because I needed to make the `dolt_docs` table mutable from the sql engine. 

We discussed Andy coming back to this later when he implements the docs for the new format

File: integration-tests/bats/keyless.bats (hunk 4 above)
Comment (druvv) [reply integration-tests/bats/keyless.bats:824]: I have an engine test covering it here:
https://github.com/dolthub/dolt/pull/3777/files#diff-51ab7d0a893924f00c2f31c46b1b61a64af3f1ea9a7a9ddda7cdf885d61094e1R1767

**Pull request merged**